x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/datafeed/DatafeedConfig.java

```diff
@@ -555,6 +600,198 @@ private TimeValue defaultFrequencyTarget(TimeValue bucketSpan) {
         return TimeValue.timeValueHours(1);
     }
 
+    static class QueryProvider implements Writeable {
```

Comment:
nit: I think readability would be improved if both providers are moved into their own files (and get corresponding unit tests)